### PR TITLE
Use Doppler v3 APIs

### PR DESCRIPTION
**Pull request by Piccirello**
This PR implements support for the Doppler v3 APIs. This also means that the CLI no longer requires specifying a project or config when using a service token.

The default fallback file path and passphrase have changed in this release. To maintain backwards compatibility, we write both the old file and the new file. Additionally, we support reading from both fallback files (with precedence given to the new file).

### CODE REVIEW

**Comment by Piccirello** [comment]
Salus error fixed in #131

**Review by bvallelunga — APPROVED** [review]

File: pkg/cmd/configs.go [hunk 1]
```diff
@@ -152,15 +149,18 @@ func deleteConfigs(cmd *cobra.Command, args []string) {
...
 	}
-	utils.RequireValue("config", config)
 
-	if yes || utils.ConfirmationPrompt("Delete config "+config, false) {
+	prompt := "Delete config"
+	if config != "" {
+		prompt = fmt.Sprintf("%s %s", prompt, config)
+	}
+
+	if yes || utils.ConfirmationPrompt(prompt, false) {
```
Comment (rgmvisser) [inline pkg/cmd/configs.go:163]: This is always true right? What is the logic behind this?

File: pkg/cmd/run.go [hunk 2]
```diff
@@ -73,18 +73,22 @@ doppler run --command "YOUR_COMMAND && YOUR_OTHER_COMMAND"`,
...
 		if cmd.Flags().Changed("fallback") {
 			var err error
 			fallbackPath, err = utils.GetFilePath(cmd.Flag("fallback").Value.String())
 			if err != nil {
 				utils.HandleError(err, "Unable to parse --fallback flag")
 			}
 		} else {
-			fallbackPath = defaultFallbackFile(localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+			fallbackPath = defaultFallbackFile(localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+			// TODO remove this when releasing CLI v4
```
Comment (rgmvisser) [inline pkg/cmd/run.go]: Can you add a task?

File: pkg/cmd/run.go [hunk 3]
```diff
@@ -285,6 +302,19 @@ func fetchSecrets(localConfig models.ScopedOptions, enableFallback bool, fallbac
 				utils.LogError(err)
 			}
 		}
+
+		// TODO remove this when releasing CLI v4
```
Comment (rgmvisser) [inline pkg/cmd/run.go]: Task pls :)

File: pkg/cmd/run.go [hunk 4]
```diff
@@ -73,18 +73,22 @@ doppler run --command "YOUR_COMMAND && YOUR_OTHER_COMMAND"`,
...
 			if err != nil {
 				utils.HandleError(err, "Unable to parse --fallback flag")
 			}
 		} else {
-			fallbackPath = defaultFallbackFile(localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+			fallbackPath = defaultFallbackFile(localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+			// TODO remove this when releasing CLI v4
+			if localConfig.EnclaveProject.Value != "" && localConfig.EnclaveConfig.Value != "" {
+				// save to old path to maintain backwards compatibility
+				legacyFallbackPath = legacyFallbackFile(localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
```
Comment (rgmvisser) [inline pkg/cmd/run.go:90]: I assume this is on the writing side? Why not read to the new path for these as well?

**Review by rgmvisser — CHANGES_REQUESTED** [review]
Couple minor comments and a small Q

File: pkg/cmd/configs.go (hunk 1 above)
Comment (Piccirello) [reply pkg/cmd/configs.go:163]: This is only true when the user specifies the `--yes` flag or they answer `yes` to the prompt

File: pkg/cmd/run.go (hunk 2 above)
Comment (Piccirello) [reply pkg/cmd/run.go]: Updated the comment

File: pkg/cmd/run.go (hunk 4 above)
Comment (Piccirello) [reply pkg/cmd/run.go:90]: We support reading from the legacy path as well in `readFallbackFile()`

File: pkg/cmd/configs.go (hunk 1 above)
Comment (rgmvisser) [reply pkg/cmd/configs.go:163]: Ah I see, somehow my hungover brain was thinking that yes in go is true :P

**Review by rgmvisser — APPROVED** [review]

**Pull request merged**